### PR TITLE
refactor simulation script and agent scripts.

### PR DESCRIPTION
**Pull request by marcofavorito**
work in progress.

### CODE REVIEW

File: simulation/v1/tac_agent_spawner.py [hunk 1]
```diff
@@ -22,20 +22,17 @@
...
+from typing import Optional, List
 
-from tac.agents.v1.base.strategy import RegisterAs, SearchFor
-from tac.agents.v1.examples.baseline import BaselineAgent
-from tac.agents.v1.examples.strategy import BaselineStrategy
-from tac.gui.monitor import VisdomMonitor, NullMonitor
-from tac.platform.controller import ControllerAgent, TACParameters
-from tac.platform.stats import GameStats
+import tac
+from tac.agents.v2.examples.baseline import BaselineAgent
```
Comment (DavidMinarsch) [inline simulation/v1/tac_agent_spawner.py]: needs to be v1

File: templates/v1/basic.py [hunk 2]
```diff
@@ -54,25 +53,6 @@ def parse_arguments():
     return parser.parse_args()
 
 
-def main():
```
Comment (DavidMinarsch) [inline templates/v1/basic.py]: I'm not a fan of this being removed. The template should make obvious what's going on. That's kind of the point of the template. So removing the strategy and agent removes the point of the template in my opinion.

File: templates/README.md [hunk 3]
```diff
@@ -34,9 +34,9 @@ If you want to use the same cryptographic key, you can follow these steps:
       
       python3 scripts/generate_private_key.py private_key.pem
       
-- Every time you run your agent, add the parameter `--private-key <pem-file>` to your command:
+- Every time you run your agent, add the parameter `--private-key-pem <pem-file>` to your command:
 
-      python3 templates/v1/basic.py --name basic1 --gui --private-key private_key.pem
+      python3 templates/v2/basic.py --name basic1 --gui --private-key-pem private_key.pem
```
Comment (DavidMinarsch) [inline templates/README.md]: v1!

File: templates/v1/basic.py (hunk 2 above)
Comment (marcofavorito) [reply templates/v1/basic.py]: Right, I'll restore the previous version

File: templates/README.md (hunk 3 above)
Comment (marcofavorito) [reply templates/README.md]: right, thanks!

**Pull request merged**